GitUI/UserControls/BlameControl.cs

```diff
@@ -332,14 +345,47 @@ private void ProcessBlame(string filename, GitRevision revision, IReadOnlyList<O
             var lineBuilder = new StringBuilder(lineLength + 2);
             var gutter = new StringBuilder(capacity: lineBuilder.Capacity * _blame.Lines.Count);
             var emptyLine = new string(' ', lineLength);
-            foreach (var line in _blame.Lines)
+            var cacheAvatars = new Dictionary<string, JoinableTask<Image>>();
+            var noAuthorTask = ThreadHelper.JoinableTaskFactory.RunAsync(() => Task.FromResult((Image)new Bitmap(Images.User80, avatarSize, avatarSize)));
+            var noAvatar = ThreadHelper.JoinableTaskFactory.RunAsync(() => Task.FromResult<Image>(null));
+            for (var index = 0; index < _blame.Lines.Count; index++)
             {
+                var line = _blame.Lines[index];
                 if (line.Commit == lastCommit)
                 {
+                    if (gitBlameDisplays != null)
+                    {
+                        gitBlameDisplays[index].Avatar = noAvatar;
+                    }
+
                     gutter.AppendLine(emptyLine);
                 }
                 else
                 {
+                    var authorEmail = line.Commit.AuthorMail?.Trim('<', '>');
+                    if (gitBlameDisplays != null)
+                    {
+                        if (authorEmail != null)
+                        {
+                            if (cacheAvatars.ContainsKey(authorEmail))
+                            {
+                                gitBlameDisplays[index].Avatar = cacheAvatars[authorEmail];
+                            }
+                            else
+                            {
+                                var avatar = ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                                    await AvatarService.Default.GetAvatarAsync(authorEmail, line.Commit.Author,
+                                        avatarSize));
```

Comment:
❓  `GetAvatarAsync` returns an image (when awaited). Why are we attempting to cache a task?

I think the following should work
```suggestion
                                Image avatar = ThreadHelper.JoinableTaskFactory.RunAsync(() => AvatarService.Default.GetAvatarAsync(authorEmail, line.Commit.Author, avatarSize));
```